Tutorials implemented

ZedXC opened the pull request:
Tutorials for all three games added
closes #12

ZedXC committed:
Tutorials implemented

Tutorials for all three games added

ZedXC committed:
Merge branch 'main' into TutorialPopups

ZedXC committed:
Made Tutorials toggle color match

tessapower commented on jest-to-impress/scripts/ui/tutorial.gd:
This is essentially a duplicate of the `submenu` scene. I would recommend removing this in favor of adding a `submenu` to each mini-game, which would reduce repetitive code and keep everything related to each individual mini-game well encapsulated. Additionally, the logic for whether to display the tutorial could then be easily handled from the mini-game base class.

The wooden backdrop looks great, so if we used it for the submenu and the pause popup scenes then the game would look cohesive! The only change we'd need to do to those scenes is add a `TextureRect`, ensure the text content is in a `ScrollContainer` and then adjust the sizes accordingly.

tessapower commented on jest-to-impress/scripts/game.gd:
```diff
@@ -35,6 +37,9 @@ func load_mini_game(idx : int) -> void:
 	mini_game_instance.connect("failure", self.on_failure)
 	mini_game_instance.connect("finish", self.on_finished)
 	add_child(mini_game_instance)
+	if GamestateManager.showTutorial and !played_games.has(idx):
```
It would be better to add this logic to the base `mini_game` class, with `has_played` as a bool flag rather than maintaining two arrays of mini games.

tessapower commented on jest-to-impress/scenes/ui/start_menu.tscn:
I'd suggest making the Tutorials checkbox font slightly bigger, changing the text to "Show Tutorials" and moving it below the "Play" button.

![image](https://github.com/tessapower/ggj-2024/assets/25911223/07051b7e-f571-41f0-8e20-1a4c7177179c)

tessapower commented on jest-to-impress/assets/graphics/Fonts/Aniron/anirb___.ttf:
I didn't realize you found a font for regular text, so went looking for one this morning! I think this font is much better though - it's super legible and we should use it for regular text and buttons, too.

tessapower reviewed (changes requested):
Looking awesome! Some changes required to tidy this up and reuse existing components, but otherwise really cool to see it all coming together so nicely :)))

ZedXC committed:
fixed issues noted in Tutorials Implemented

-Reuses the submenu class
- Potential fix for the minigame thing. This was the only way I could think to do it gotta ask Tessa about this. Will stay up all night.
- Made text on game.tscn bigger

ZedXC committed:
Checks if tutorials are on or off

ZedXC committed:
Scroll buttons for the submenu added

ZedXC committed:
Changed Pause Menu

ZedXC committed:
Made project not change size

tessapower committed:
Adjust dimensions of pause popup

tessapower committed:
Adjust dimensions and add scroll container to submenu

tessapower committed:
Remove unused tutorial scene and script

tessapower committed:
Refactor logic for showing tutorials in mini-games

tessapower committed:
Add submenu for tutorial popups to mini-games

tessapower committed:
Minor tidyup

tessapower committed:
Merge branch 'main' into TutorialPopups
